### PR TITLE
Add notification manager API

### PR DESCRIPTION
**Pull request by olexii4**
Add notification manager API to the new plugin system.
Discussion about architecture and principles can be found here:  https://github.com/theia-ide/theia/issues/1482
This API allows implementation of notifications to show an information message. Optionally provide an array of items which will be presented as clickable buttons and can be shown using the [showInformationMessage](#window.showInformationMessage),
 [showWarningMessage](#window.showWarningMessage) and [showErrorMessage](#window.showErrorMessage) functions.

Simple example that show an information message:
```
theia.window.showInformationMessage('Information message');
```
Simple example that show an information message with buttons:
```
theia.window.showInformationMessage('Information message', 'Btn1', 'Btn2').then(result => {
    console.log("Click button", result);
});
```
https://github.com/eclipse/che/issues/9525

selection_141
selection_148
selection_154
Both kind of notifications with buttons:
selection_136

Signed-off-by: Oleksii Orel <oorel@redhat.com>

### CODE REVIEW

File: packages/plugin/src/api/plugin-api.ts [hunk 1]
```diff
@@ -70,11 +82,13 @@ export interface QuickOpenMain {
...
+    QUICK_OPEN_MAIN: createProxyIdentifier<QuickOpenMain>('QuickOpenMain'),
+    MESSAGE_REGISTRY_MAIN: <ProxyIdentifier<MessageRegistryMain>>createProxyIdentifier<MessageRegistryMain>('MessageRegistryMain')
 };
 
 export const MAIN_RPC_CONTEXT = {
     HOSTED_PLUGIN_MANAGER_EXT: createProxyIdentifier<HostedPluginManagerExt>('HostedPluginManagerExt'),
     COMMAND_REGISTRY_EXT: createProxyIdentifier<CommandRegistryExt>('CommandRegistryExt'),
-    QUICK_OPEN_EXT: createProxyIdentifier<QuickOpenExt>('QuickOpenExt')
+    QUICK_OPEN_EXT: createProxyIdentifier<QuickOpenExt>('QuickOpenExt'),
+    MESSAGE_REGISTRY_EXT: createProxyIdentifier('MessageRegistryExt')
```
Comment (evidolob) [inline packages/plugin/src/api/plugin-api.ts]: Why you add this `MESSAGE_REGISTRY_EXT` ? Is this used anywhere?

Comment (olexii4) [reply packages/plugin/src/api/plugin-api.ts]: I used it in plugin-context.ts
https://github.com/theia-demo-plugins/wiptheia/pull/2/files#diff-761ce14ea07b1d3bb10a3b02a8fb1763R22

File: packages/plugin/src/plugin/plugin-context.ts [hunk 2]
```diff
@@ -14,10 +14,12 @@ import { CancellationTokenSource } from '@theia/core/lib/common/cancellation';
 import { QuickOpenExtImpl } from './quick-open';
 import { Plugin } from '../api/plugin-api';
 import { getPluginId } from '../common/plugin-protocol';
+import { MessageRegistryExt } from './message-registry';
 
 export function createAPI(rpc: RPCProtocol): typeof theia {
     const commandRegistryExt = rpc.set(MAIN_RPC_CONTEXT.COMMAND_REGISTRY_EXT, new CommandRegistryImpl(rpc));
     const quickOpenExt = rpc.set(MAIN_RPC_CONTEXT.QUICK_OPEN_EXT, new QuickOpenExtImpl(rpc));
+    const messageRegistryExt = rpc.set(MAIN_RPC_CONTEXT.MESSAGE_REGISTRY_EXT, new MessageRegistryExt(rpc));
```
Comment (evidolob) [inline packages/plugin/src/plugin/plugin-context.ts]: You don't need to do `rpc.set`, as your `MessageRegistryExt` is never called from Theia core part

File: packages/plugin/src/api/plugin-api.ts (hunk 1 above)
Comment (evidolob) [reply packages/plugin/src/api/plugin-api.ts]: It's unnecessary, see comment below

File: packages/plugin/src/plugin/plugin-context.ts (hunk 2 above)
Comment (olexii4) [reply packages/plugin/src/plugin/plugin-context.ts]: Ok. I just added a mechanism for possible future In the first realization. I'll remove.

Comment (olexii4) [reply packages/plugin/src/plugin/plugin-context.ts]: done

**Review by evidolob — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Review by evidolob — APPROVED** [review]

File: packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+.dialogContent .theia-Notification {
+    min-width: inherit;
+    box-shadow: none;
+    animation: none;
+}
+
+.dialogContent .theia-Notification .icon {
+    font-size: 20px;
+    padding: 5px 0;
+}
```
Comment (benoitf) [inline packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css]: there is a missing EOF

Comment (olexii4) [reply packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css]: done

**Review by benoitf — APPROVED** [review]

**Pull request merged**